flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/ZookeeperOffsetHandler.java

```diff
@@ -98,30 +97,42 @@ public void seekFetcherToInitialOffsets(List<KafkaTopicPartitionLeader> partitio
 
 	@Override
 	public void close() throws IOException {
-		zkClient.close();
+		curatorClient.close();
 	}
 
 	// ------------------------------------------------------------------------
 	//  Communication with Zookeeper
 	// ------------------------------------------------------------------------
 	
-	public static void setOffsetInZooKeeper(ZkClient zkClient, String groupId, String topic, int partition, long offset) {
-		TopicAndPartition tap = new TopicAndPartition(topic, partition);
-		ZKGroupTopicDirs topicDirs = new ZKGroupTopicDirs(groupId, tap.topic());
-		ZkUtils.updatePersistentPath(zkClient, topicDirs.consumerOffsetDir() + "/" + tap.partition(), Long.toString(offset));
+	public static void setOffsetInZooKeeper(CuratorFramework curatorClient, String groupId, String topic, int partition, long offset) throws Exception {
+		ZKGroupTopicDirs topicDirs = new ZKGroupTopicDirs(groupId, topic);
+		String path = topicDirs.consumerOffsetDir() + "/" + partition;
+		ensureExists(curatorClient, path);
+		byte[] data = Long.toString(offset).getBytes();
```

Comment:
Depending on how often this is called, we could replace it with another way of converting it to the original bytes.